rules/avoid-deleting-records-by-flagging-them-as-isdeleted/rule.md

```diff
@@ -40,18 +35,77 @@ Disadvantages
 * Depending on your interface design, you may have to join to parent tables to ensure that deleted child records do not appear. Typically, the interface would be designed in such a way that you would not need be able to created new records based on the deleted items (e.g. you cannot create a new order record for a customer that is deleted). Performance of queries can potentially suffer if you have to do these joins.
 * While storage space is very cheap, you are not removing records from your database. You may need to archive records if the number of deleted records becomes large.
 
+### Best Approach for Implementing Soft Delete in EF Core for modern web application.
 
-Also see [Using Audit Tools](/use-temporal-tables-to-audit-data-changes) for alternatives to this approach using 3rd party auditing tools.
+### 1. `IsSoftDeleteEntity` Interface
```

Comment:
Interfaces always start with an `I` in addition to their name. Suggested name:
```suggestion
### 1. `ISoftDeleteEntity` Interface
```

Usually, we would use it in combination of auditable tables by that case we probably wouldn't use a separate interface. This would be best approach, while just using `ISoftDeleteEntity` would be simplest way to add where auditing isn't required.

``` csharp
public interface IAuditEntity
{
    DateTime CreatedOn { get; set; }
    string CreatedBy { get; set; }

    DateTime UpdatedOn { get; set; }
    string UpdatedBy { get; set; }

    DateTime? DeletedOn { get; set; }
    string DeletedBy { get; set; }

    public bool IsDeleted { get; set; }
}
```

Some would use only `DeletedOn != null` for determining if it's deleting, so `IsDeleted` is optional. It still look nicer `IsDeleted`.

Reply:
Let's put a pin on the "Best approach" and keep it as is, just change from `IsSoftDeleteEntity` to `ISoftDeleteEntity`.
We should see if we have proper audit table tracking for EF Core and link it to that as the actual best approach and this is still great approach if audit fields are not needed (usually they are when you start adding soft delete)